main.go

```diff
@@ -440,6 +442,7 @@ func handlePanicWhileMounting() {
 	}
 }
 
+//go:generate go run codegen/main.go --inF=codegen/params.yaml --outF=cfg/config.go --tmplF=codegen/config.tpl
```

Comment:
What is this comment for? If this is the command for code generation, shall we move it to the appropriate package? Also let's add a comment to clarify what this is used for.

Reply:
This comment is special: when the `go generate` command is triggered from the command line, golang searches through the go files for comments of this type i.e. //go:generate <shell_command> and triggers the shell_command. So, this comment is required for generating the config.go.

Reply:
Got it, let's add a comment explaining this so that we know in future why this was added